<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Denny Shin" />

<meta name="date" content="2018-09-26" />

<title>4kpbmc</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Learning. Thoughts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">4kpbmc</h1>
<h4 class="author"><em>Denny Shin</em></h4>
<h4 class="date"><em>2018-09-26</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-09-26</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180724)</code> </summary></p>
<p>The command <code>set.seed(20180724)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/dennyshin/Learning-Thoughts/tree/6bc7634f62bde879bdd94383ddb2c2af2e3462d7" target="_blank">6bc7634</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    docs/figure/

Untracked files:
    Untracked:   (DESKTOP-9MQETIV's conflicted copy 2018-07-26).Rhistory
    Untracked:  Learning-Thoughts.Rproj
    Untracked:  data/4k PBMCs from a Healthy Donor/
    Untracked:  data/filtered_gene_bc_matrices/
    Untracked:  output/pbmc3k_final.rds
    Untracked:  output/pbmc_tutorial.rds

Unstaged changes:
    Deleted:    Learning. Thoughts.Rproj
    Modified:   analysis/long-read-seq-CACNA1C.Rmd
    Modified:   analysis/single-cell-RNAseq.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/6bc7634f62bde879bdd94383ddb2c2af2e3462d7/analysis/4kpbmc.Rmd" target="_blank">6bc7634</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-09-26
</td>
<td style="text-align:left;">
added 4kpbmc and zombie lif article
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="introduction" class="section level1">
<h1><strong>Introduction</strong></h1>
<p>4k PBMC single cell analysis</p>
<p>I will follow along with Seurat</p>
</div>
<div id="setup-the-seurat-object" class="section level1">
<h1><strong>Setup the Seurat Object</strong></h1>
<pre class="r"><code>library(Seurat)</code></pre>
<pre><code>Warning: package &#39;Seurat&#39; was built under R version 3.5.1</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 3.5.1</code></pre>
<pre><code>Loading required package: cowplot</code></pre>
<pre><code>Warning: package &#39;cowplot&#39; was built under R version 3.5.1</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 3.5.1</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>pbmc.data = Read10X(data.dir = &quot;../data/4k PBMCs from a Healthy Donor/pbmc4k_filtered_gene_bc_matrices/filtered_gene_bc_matrices/GRCh38/&quot;)

dense.size = object.size(x =  as.matrix(x = pbmc.data))
dense.size</code></pre>
<pre><code>1172516744 bytes</code></pre>
<pre class="r"><code>sparse.size = object.size(x = pbmc.data)
sparse.size</code></pre>
<pre><code>94305296 bytes</code></pre>
<pre class="r"><code>dense.size/sparse.size</code></pre>
<pre><code>12.4 bytes</code></pre>
<pre class="r"><code>pbmc &lt;- CreateSeuratObject(raw.data = pbmc.data, min.cells = 3, min.genes = 200, project = &quot;10X_4kPBMC&quot;)</code></pre>
</div>
<div id="qc" class="section level1">
<h1><strong>QC</strong></h1>
<div id="filter-poor-quality-cells" class="section level4">
<h4><strong>filter poor quality cells</strong></h4>
<pre class="r"><code>mito.genes &lt;- grep(pattern = &quot;^MT-&quot;, x = rownames(x = pbmc@data), value = TRUE)
percent.mito &lt;- Matrix::colSums(pbmc@raw.data[mito.genes, ])/Matrix::colSums(pbmc@raw.data)

pbmc &lt;- AddMetaData(object = pbmc, metadata = percent.mito, col.name = &quot;percent.mito&quot;)
VlnPlot(object = pbmc, features.plot = c(&quot;nGene&quot;, &quot;nUMI&quot;, &quot;percent.mito&quot;), nCol = 3)</code></pre>
<p><img src="figure/4kpbmc.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1, 2))
GenePlot(object = pbmc, gene1 = &quot;nUMI&quot;, gene2 = &quot;percent.mito&quot;)
GenePlot(object = pbmc, gene1 = &quot;nUMI&quot;, gene2 = &quot;nGene&quot;)</code></pre>
<p><img src="figure/4kpbmc.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>from the graphs, I decided to filter out cells with 10 or more percentage of mitochondrial genes</p>
<p>also cells with unique gene counts over 4000</p>
<pre class="r"><code>pbmc &lt;- FilterCells(object = pbmc, subset.names = c(&quot;nGene&quot;, &quot;percent.mito&quot;), low.thresholds = c(200, -Inf), high.thresholds = c(4000, 0.10))</code></pre>
</div>
<div id="normalising-the-data" class="section level4">
<h4><strong>Normalising the data</strong></h4>
<pre class="r"><code>pbmc &lt;- NormalizeData(object = pbmc, normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000)</code></pre>
</div>
</div>
<div id="detection-of-variable-genes-across-the-single-cells" class="section level1">
<h1><strong>Detection of variable genes across the single cells</strong></h1>
<p>The parameters chosen here should most likely be similar to the Seurat tutorial but there may be some things I am missing. What are some good reasons to help decide on good cutoff values?</p>
<pre class="r"><code>pbmc &lt;- FindVariableGenes(object = pbmc, mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, x.high.cutoff = 3, y.cutoff = 0.5)</code></pre>
<p><img src="figure/4kpbmc.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>length(x = pbmc@var.genes)</code></pre>
<pre><code>[1] 1680</code></pre>
</div>
<div id="scaling-data-and-removing-unwanted-sources-of-variation" class="section level1">
<h1><strong>Scaling data and removing unwanted sources of variation</strong></h1>
<pre class="r"><code>pbmc &lt;- ScaleData(object = pbmc, vars.to.regress = c(&quot;nUMI&quot;, &quot;percent.mito&quot;))</code></pre>
<pre><code>Regressing out: nUMI, percent.mito</code></pre>
<pre><code>
Time Elapsed:  13.0137679576874 secs</code></pre>
<pre><code>Scaling data matrix</code></pre>
</div>
<div id="perform-linear-dimensional-reduction" class="section level1">
<h1><strong>Perform linear dimensional reduction</strong></h1>
<pre class="r"><code>pbmc &lt;- RunPCA(object = pbmc, pc.genes = pbmc@var.genes, do.print = TRUE, pcs.print = 1:5, genes.print = 5)</code></pre>
<pre><code>[1] &quot;PC1&quot;
[1] &quot;IL32&quot; &quot;CD3E&quot; &quot;LTB&quot;  &quot;CD7&quot;  &quot;CTSW&quot;
[1] &quot;&quot;
[1] &quot;CSTA&quot; &quot;MNDA&quot; &quot;FCN1&quot; &quot;CST3&quot; &quot;LST1&quot;
[1] &quot;&quot;
[1] &quot;&quot;
[1] &quot;PC2&quot;
[1] &quot;CD79A&quot; &quot;MS4A1&quot; &quot;IGHM&quot;  &quot;CD79B&quot; &quot;IGHD&quot; 
[1] &quot;&quot;
[1] &quot;IL32&quot; &quot;CTSW&quot; &quot;NKG7&quot; &quot;CD3E&quot; &quot;GZMA&quot;
[1] &quot;&quot;
[1] &quot;&quot;
[1] &quot;PC3&quot;
[1] &quot;NKG7&quot;   &quot;GZMB&quot;   &quot;PRF1&quot;   &quot;FGFBP2&quot; &quot;CST7&quot;  
[1] &quot;&quot;
[1] &quot;LEF1&quot; &quot;MAL&quot;  &quot;LTB&quot;  &quot;CCR7&quot; &quot;IL7R&quot;
[1] &quot;&quot;
[1] &quot;&quot;
[1] &quot;PC4&quot;
[1] &quot;LILRA4&quot; &quot;LRRC26&quot; &quot;CLEC4C&quot; &quot;PTCRA&quot;  &quot;TPM2&quot;  
[1] &quot;&quot;
[1] &quot;FCGR3A&quot; &quot;CD79B&quot;  &quot;IGHD&quot;   &quot;MS4A1&quot;  &quot;CD79A&quot; 
[1] &quot;&quot;
[1] &quot;&quot;
[1] &quot;PC5&quot;
[1] &quot;LILRA4&quot;   &quot;SERPINF1&quot; &quot;CLEC4C&quot;   &quot;LRRC26&quot;   &quot;SCT&quot;     
[1] &quot;&quot;
[1] &quot;SDPR&quot;  &quot;PPBP&quot;  &quot;PF4&quot;   &quot;GNG11&quot; &quot;TUBB1&quot;
[1] &quot;&quot;
[1] &quot;&quot;</code></pre>
<pre class="r"><code>VizPCA(object = pbmc, pcs.use = 1:2)</code></pre>
<p><img src="figure/4kpbmc.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VizPCA(object = pbmc, pcs.use = 1:2, num.genes = 10)</code></pre>
<p><img src="figure/4kpbmc.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>PCAPlot(object = pbmc)</code></pre>
<p><img src="figure/4kpbmc.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can notice 2 or maybe 3 different cell types??</p>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17758)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_0.7.6   Seurat_2.3.4  Matrix_1.2-14 cowplot_0.9.3 ggplot2_3.0.0
[6] knitr_1.20   

loaded via a namespace (and not attached):
  [1] Rtsne_0.13          colorspace_1.3-2    class_7.3-14       
  [4] modeltools_0.2-22   ggridges_0.5.0      mclust_5.4.1       
  [7] rprojroot_1.3-2     htmlTable_1.12      base64enc_0.1-3    
 [10] rstudioapi_0.7      proxy_0.4-22        flexmix_2.3-14     
 [13] bit64_0.9-7         mvtnorm_1.0-8       codetools_0.2-15   
 [16] splines_3.5.0       R.methodsS3_1.7.1   robustbase_0.93-2  
 [19] Formula_1.2-3       jsonlite_1.5        workflowr_1.1.1    
 [22] ica_1.0-2           cluster_2.0.7-1     kernlab_0.9-27     
 [25] png_0.1-7           R.oo_1.22.0         compiler_3.5.0     
 [28] httr_1.3.1          backports_1.1.2     assertthat_0.2.0   
 [31] lazyeval_0.2.1      lars_1.2            acepack_1.4.1      
 [34] htmltools_0.3.6     tools_3.5.0         bindrcpp_0.2.2     
 [37] igraph_1.2.2        gtable_0.2.0        glue_1.3.0         
 [40] RANN_2.6            reshape2_1.4.3      Rcpp_0.12.18       
 [43] trimcluster_0.1-2.1 gdata_2.18.0        ape_5.1            
 [46] nlme_3.1-137        iterators_1.0.10    fpc_2.1-11.1       
 [49] gbRd_0.4-11         lmtest_0.9-36       stringr_1.3.1      
 [52] irlba_2.3.2         gtools_3.8.1        DEoptimR_1.0-8     
 [55] MASS_7.3-49         zoo_1.8-3           scales_1.0.0       
 [58] doSNOW_1.0.16       parallel_3.5.0      RColorBrewer_1.1-2 
 [61] yaml_2.2.0          reticulate_1.10     pbapply_1.3-4      
 [64] gridExtra_2.3       rpart_4.1-13        segmented_0.5-3.0  
 [67] latticeExtra_0.6-28 stringi_1.1.7       foreach_1.4.4      
 [70] checkmate_1.8.5     caTools_1.17.1.1    bibtex_0.4.2       
 [73] Rdpack_0.9-0        SDMTools_1.1-221    rlang_0.2.2        
 [76] pkgconfig_2.0.2     dtw_1.20-1          prabclus_2.2-6     
 [79] bitops_1.0-6        evaluate_0.11       lattice_0.20-35    
 [82] ROCR_1.0-7          purrr_0.2.5         bindr_0.1.1        
 [85] labeling_0.3        htmlwidgets_1.2     bit_1.1-14         
 [88] tidyselect_0.2.4    plyr_1.8.4          magrittr_1.5       
 [91] R6_2.2.2            snow_0.4-2          gplots_3.0.1       
 [94] Hmisc_4.1-1         pillar_1.3.0        whisker_0.3-2      
 [97] foreign_0.8-70      withr_2.1.2         fitdistrplus_1.0-9 
[100] mixtools_1.1.0      survival_2.41-3     nnet_7.3-12        
[103] tsne_0.1-3          tibble_1.4.2        crayon_1.3.4       
[106] hdf5r_1.0.0         KernSmooth_2.23-15  rmarkdown_1.10     
[109] grid_3.5.0          data.table_1.11.4   git2r_0.23.0       
[112] metap_1.0           digest_0.6.16       diptest_0.75-7     
[115] tidyr_0.8.1         R.utils_2.7.0       stats4_3.5.0       
[118] munsell_0.5.0      </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
