<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Denny Shin" />

<meta name="date" content="2018-07-25" />

<title>eQTL discovery in adipose tissue</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Learning. Thoughts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">eQTL discovery in adipose tissue</h1>
<h4 class="author"><em>Denny Shin</em></h4>
<h4 class="date"><em>2018-07-25</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-07-31</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/dennyshin/Learning-Thoughts/tree/256cb8b2264113d4a9357d1e5a54ca44e942fad6" target="_blank">256cb8b</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:   (DESKTOP-9MQETIV's conflicted copy 2018-07-26).Rhistory

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/256cb8b2264113d4a9357d1e5a54ca44e942fad6/analysis/eQTL-adipose-tissue.Rmd" target="_blank">256cb8b</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-07-31
</td>
<td style="text-align:left;">
i think i have finished a first draft on cibersort and adipose tissue pages
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/dennyshin/Learning-Thoughts/713321cdf715e1d5fda2e24e670655c492fc98ca/docs/eQTL-adipose-tissue.html" target="_blank">713321c</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-07-30
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/594ef3d5f254adb25ae703050d59d2fda08014bc/analysis/eQTL-adipose-tissue.Rmd" target="_blank">594ef3d</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-07-30
</td>
<td style="text-align:left;">
more stuff in adipose tissue page
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/dennyshin/Learning-Thoughts/d4b6a3e60a6cd4619189371c677faf5667a1ef4e/docs/eQTL-adipose-tissue.html" target="_blank">d4b6a3e</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-07-30
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/e3ab44c041b82ed64b69c16a0ac9a5844cb1c4c0/analysis/eQTL-adipose-tissue.Rmd" target="_blank">e3ab44c</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-07-30
</td>
<td style="text-align:left;">
added CIBERSORT page. Continued on the adipose tissue page.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/dennyshin/Learning-Thoughts/b69d2301affdfa63e58ef584b4d845e2d94b2a7d/docs/eQTL-adipose-tissue.html" target="_blank">b69d230</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-07-25
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/1947d8ad6325431f4ca4c957b2a1b30328e9e05c/analysis/eQTL-adipose-tissue.Rmd" target="_blank">1947d8a</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-07-25
</td>
<td style="text-align:left;">
created two new pages. finally made articles 2018 page. dont forget to change my site title if i even can.
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="introduction" class="section level2">
<h2><strong>Introduction</strong></h2>
<p>A summary of an article about eQTL discovery using CIBERSORT.</p>
<p>My main motivation was to learn how they approached the eQTL analysis and their use of CIBERSORT.</p>
</div>
<div id="testing-ciberseort" class="section level2">
<h2><strong>Testing CIBERSEORT</strong></h2>
<p>The first step was to estimate the cell type proportions using RNAseq datasets. But before jumping right into the estimation, CIBERSORT’s accuracy and robustness were tested against well-defined data sets. The first test showed that CIBERSORT can, with a high degree of accuracy, identify cell types in purified cell type RNA-seq data. In the second test, 1000 simulated cell type mixtures were accurately identified by CIBERSORT. The third test involved the same simulations but with unknown cell types (they added two new cell types to the original four). CIBERSORT predicted accurately when the unknown cells were less than 10% of the total cell composition (the article mentions that adipose tissue is likely to be less than 5% of unknown cell types). Finally, the last test was to add Gaussian noise (at 10, 50 and 90%). CIBERSORT was accurate with up to 10% noise.</p>
</div>
<div id="estimating-cell-type-proportions" class="section level2">
<h2><strong>Estimating cell type proportions</strong></h2>
<p>CIBERSORT was applied to a dataset obtained from TwinsUK (these samples successfully deconvulved at an FDR 1%). These results also agreed with previous studies using flow cytometry (an older method of determining cell-type proportions). Another dataset, this time from GTEx, was applied to CIBERSORT and showed a much less successful deconvolution rate (~23% failed). Comparing the two datasets (only those that successfully deconvulved) GTEx samples had:</p>
<ul>
<li>lower adipocyte fraction</li>
<li>two times more vasculature</li>
<li>4 times more macrophages</li>
</ul>
<p>They suspect that the lower expression of ADIPOQ in the GTEx are whats causing the degraded data. The article lists more potential causes but I don’t really follow the biology.</p>
</div>
<div id="adipose-cell-types-are-heritable" class="section level2">
<h2><strong>Adipose cell-types are heritable</strong></h2>
<p>The article mentions that they use classical twin models to determine hertibility. It seems that this method, in a sense, is making use of identical twins to find genetic traits. The article also mentions that certain physiological factors may mediate the adipose cell composition.</p>
</div>
<div id="adjusting-for-macrophage-heterogeneity" class="section level2">
<h2><strong>Adjusting for Macrophage heterogeneity</strong></h2>
<p>It was found that macrophage proportion was significantly affecting the result, association with BMI and adipose composition. I sort of get what they’re doing here but I’m not really following why.</p>
</div>
<div id="gene-expression-variance-and-co-variance" class="section level2">
<h2><strong>Gene expression variance and co-variance</strong></h2>
<p>There’s some detail about how cell-type proportion influences gene expression in adipose tissue but it’s hard to tell why without the PCA plot or from their explanation.</p>
</div>
<div id="cis-eqtl-discovery" class="section level2">
<h2><strong>cis-eQTL discovery</strong></h2>
<p>There is an increase in the discovery of cis-eQTL when cell-type proportions are included in the analysis. Article mentions the use of probabilistic estimation of expression residuals (PEER) factors to adjust the eQTL analysis. More on this should be read.</p>
</div>
<div id="additional-thoughts" class="section level2">
<h2><strong>Additional Thoughts</strong></h2>
<p>There seems to be a lot to learn about the “correct” approaches to a bioinformatic study from this article that I can learn. Some statistical jargon and methods are way above my head at this level but I should be able to fill them in. It was also difficult to follow the article at times when so much detail about the biology was written that I have little knowledge of.</p>
</div>
<div id="links-and-information" class="section level2">
<h2><strong>Links and Information</strong></h2>
<p>Article Name: Cell-type heterogeneity in adipose tissue is associated with complex traits and reveals disease-relevant cell-specific eQTLs</p>
<p>Main Author: Craig A Glastonbury</p>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
