<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Denny Shin" />

<meta name="date" content="2018-08-08" />

<title>long-read-seq-CACNA1C</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Learning. Thoughts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">long-read-seq-CACNA1C</h1>
<h4 class="author"><em>Denny Shin</em></h4>
<h4 class="date"><em>2018-08-08</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-08-08</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/dennyshin/Learning-Thoughts/tree/c353465c9f7014962393202d3f7e0064a3291016" target="_blank">c353465</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:   (DESKTOP-9MQETIV's conflicted copy 2018-07-26).Rhistory
    Untracked:  Learning-Thoughts.Rproj

Unstaged changes:
    Deleted:    Learning. Thoughts.Rproj

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/c353465c9f7014962393202d3f7e0064a3291016/analysis/long-read-seq-CACNA1C.Rmd" target="_blank">c353465</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-08-08
</td>
<td style="text-align:left;">
added long read seq in cacna1c article.
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="motivation" class="section level2">
<h2><strong>Motivation</strong></h2>
<p>We focus on the impact of RNA splicing on schizophrenia, more specifically, the CACNA1C gene. Why schizophrenia? Well, because schizophrenia is the disorder which most genetic loci have been identified to date (Feb 2018). Short-read RNAseq misses out on alot of the gene and splice isoforms. Also, because of read alignment it is hard to distinguish between full-length isoforms. Our motivation is to use long-red RNAseq to improve RNA splice isoform discovery.</p>
</div>
<div id="cacna1c-gene" class="section level2">
<h2><strong>CACNA1C gene</strong></h2>
<p>CACNA1C encodes a voltage-gated calcium channel (VGCC) and is a leading candidate gene forpsychiatric disorders. We currently (Feb 2018) observe ~50 exons and 40 isoforms (from splicing and transcriptional mechanisms). Alternative forms of the gene shows functionally different VGCCs.</p>
</div>
<div id="method" class="section level2">
<h2><strong>Method</strong></h2>
<p>Oxford nanopore will be used to provide the long-read sequences while bearing in mind that these reads have higher than normal error rates. To compensate, RT-PCR is combined with nanopore technolofy.</p>
<p>There is a lot of terminology that I don’t understand but I think I get the general idea of what they’re doing. After RNA extraction (which I have no clue about) they do RT-PCR followed by nanopore sequencing. They map the reads and annotate transcripts. From this they find the expression levels. They also find new exons and exon junctions (the joining place of two exons on the mature mRNA). Lastly, they mention the effect of amplification of full-length genes but I don’t really follow. Note: RIN = RNA integrity number.</p>
</div>
<div id="results" class="section level2">
<h2><strong>Results</strong></h2>
<div id="long-range-seq-reveal-new-things" class="section level4">
<h4><strong>Long-range seq reveal new things</strong></h4>
<p>Specifically, gene isoforms and exons. The new exons were validated using PCR and Sanger sequencing. Results also show high expression of a previously known transcript as expected. It validates the new exons and isoforms and also suggests that current annotations are missing many or represent repeats/errors.</p>
</div>
<div id="impact-on-protein-sequence" class="section level4">
<h4><strong>Impact on protein sequence</strong></h4>
<p>Recall: CACNA1C encodes a VGCC. VGCC contains a pore, which are made of transmembrane repeats. It is known that disruptions to these membrane lead to a non-functioning protein. Thus, they looked for isoforms where the membrane coding regions are impacted. Long read sequencing has proved useful once again by being able to reliably identify microindels (given we have enough reads). The article also talks about the impact on the differen domains and channel properties of the protein.</p>
</div>
<div id="expression-across-different-brain-regions-tissues" class="section level4">
<h4><strong>Expression across different brain regions (tissues)</strong></h4>
<p>They found that individuals had no impact on gene expression but brain tissue did. Different tissues also showed different dominance of gene isoforms i.e. different tissues had different proportions of gene isoforms.</p>
</div>
<div id="problems-with-long-red-sequencing" class="section level4">
<h4><strong>Problems with long-red sequencing</strong></h4>
<p>High quality of RNA is needed for RT-PCR and long-read sequences. Post-mortem samples are often degraded. Article suggest a minimum RINe (?) value of 6 or 7 depending on intended use.</p>
</div>
</div>
<div id="my-thoughts" class="section level2">
<h2><strong>My thoughts</strong></h2>
<p>So, always take my own thoughts with a full bag of salt because I know much less than any of these people. It is nice that we identified more transcripts but how do we compare the quality of these findings versus other methods? The article suggests very boldly that many other isoforms are repeats and errors but I am not really convinced. There is good reason to think this but I think that further testing and improvement of long-read sequences should be done before we reach this conclusion. Regardless, it is still exciting to think that long-reads are not far away from being used reliably for a variety of studies.</p>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
