<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Denny Shin" />

<meta name="date" content="2018-08-13" />

<title>Seurat</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Learning. Thoughts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Seurat</h1>
<h4 class="author"><em>Denny Shin</em></h4>
<h4 class="date"><em>2018-08-13</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-09-18</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180724)</code> </summary></p>
<p>The command <code>set.seed(20180724)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/dennyshin/Learning-Thoughts/tree/b42a4321e90b6e1e69d91691a011ebc854e7228b" target="_blank">b42a432</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:   (DESKTOP-9MQETIV's conflicted copy 2018-07-26).Rhistory
    Untracked:  Learning-Thoughts.Rproj
    Untracked:  data/filtered_gene_bc_matrices/
    Untracked:  output/pbmc3k_final.rds
    Untracked:  output/pbmc_tutorial.rds

Unstaged changes:
    Deleted:    Learning. Thoughts.Rproj
    Modified:   analysis/long-read-seq-CACNA1C.Rmd
    Modified:   analysis/single-cell-RNAseq.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/b42a4321e90b6e1e69d91691a011ebc854e7228b/analysis/Seurat.Rmd" target="_blank">b42a432</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-09-18
</td>
<td style="text-align:left;">
wflow_publish(“analysis/Seurat.Rmd”)
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/dennyshin/Learning-Thoughts/f97dbbb50fa808e441dc0f0440a86dc5664e18ec/docs/Seurat.html" target="_blank">f97dbbb</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-08-14
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/0378f6512d52f10716172d3fa3b85d565865d2b8/analysis/Seurat.Rmd" target="_blank">0378f65</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-08-14
</td>
<td style="text-align:left;">
added seurat page, expanded on single cell rnaseq
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="introduction" class="section level2">
<h2><strong>Introduction</strong></h2>
<p>I will be following through the code written <a href="https://satijalab.org/seurat/pbmc3k_tutorial.html">here</a></p>
<p>The goal of this tutorial is to demonstrate clustering of RNAseq data. Clustering is the process of grouping similar objects together.</p>
<p>It uses the packages Seurat and dplyr. The Seurat package takes a while.</p>
</div>
<div id="setup-the-seurat-object" class="section level2">
<h2><strong>Setup the Seurat Object</strong></h2>
<p>Sparse matrices are matrices that contain mostly zeros.</p>
<pre class="r"><code>library(Seurat)
library(dplyr)

# Load the PBMC dataset
pbmc.data &lt;- Read10X(data.dir = &quot;../data/filtered_gene_bc_matrices/hg19/&quot;)

# Examine the memory savings between regular and sparse matrices
dense.size &lt;- object.size(x = as.matrix(x = pbmc.data))
dense.size

sparse.size &lt;- object.size(x = pbmc.data)
sparse.size

dense.size/sparse.size</code></pre>
<p>This first bit is to demonstrate the memory saving capability of Seurat.</p>
<pre class="r"><code># Initialize the Seurat object with the raw (non-normalized data). Keep all genes expressed in &gt;= 3 cells (~0.1% of the data). Keep all cells with at least 200 detected genes
pbmc &lt;- CreateSeuratObject(raw.data = pbmc.data, min.cells = 3, min.genes = 200, project = &quot;10X_PBMC&quot;)</code></pre>
<p>In <a href="mailto:pbmc@meta.data">pbmc@meta.data</a> it contains the nGene and nUMI data for each cell.</p>
</div>
<div id="qc-and-selecting-cells-for-further-analysis" class="section level2">
<h2><strong>QC and selecting cells for further analysis</strong></h2>
<p>CreateSeuratObject can do a basic cutoff for minimum genes but for more sophisticated filtering (based on technical or biological parameters) we can do the following. Seurat is flexible in allowing user-specific filtering methods. An example:</p>
<p>In this process we try to exclude “multiplets”, a phenomenom where multiple cells are in a single droplet. If there is one cell per one droplet, the number of genes observed should reflect the number of UMIs. We also filter cells based on the percentage of mitochondrial genes. High proportions of mitochondrial genome are indicative of poor-quality cells. I should find some published material for this.</p>
<pre class="r"><code>mito.genes &lt;- grep(pattern = &quot;^MT-&quot;, x = rownames(x = pbmc@data), value = TRUE)
percent.mito &lt;- Matrix::colSums(pbmc@raw.data[mito.genes, ])/Matrix::colSums(pbmc@raw.data)

pbmc &lt;- AddMetaData(object = pbmc, metadata = percent.mito, col.name = &quot;percent.mito&quot;)
VlnPlot(object = pbmc, features.plot = c(&quot;nGene&quot;, &quot;nUMI&quot;, &quot;percent.mito&quot;), nCol = 3)</code></pre>
<p>VlnPlot is built in to Seurat.</p>
<p>Take a closer look into what we just did:</p>
<pre class="r"><code>dim(pbmc@raw.data)
pbmc@raw.data[mito.genes, ]</code></pre>
<p>Ok, so we have 13714 genes (row) and 2700 cells (col). The names of the cells confuse me, are they from the cDNA barcode?</p>
<pre class="r"><code>par(mfrow = c(1, 2))
GenePlot(object = pbmc, gene1 = &quot;nUMI&quot;, gene2 = &quot;percent.mito&quot;)
GenePlot(object = pbmc, gene1 = &quot;nUMI&quot;, gene2 = &quot;nGene&quot;)</code></pre>
<p>par() sets graphical parameters in R.</p>
<p>Looking at the plots we can identify the outliers and set cutoffs. Here, we set a cutoff of 5% mitochondrial genes. Roger mentioned that typically 10% is used.</p>
<pre class="r"><code>pbmc &lt;- FilterCells(object = pbmc, subset.names = c(&quot;nGene&quot;, &quot;percent.mito&quot;), low.thresholds = c(200, -Inf), high.thresholds = c(2500, 0.05))</code></pre>
</div>
<div id="normalizing-the-data" class="section level2">
<h2><strong>Normalizing the data</strong></h2>
<pre class="r"><code>pbmc &lt;- NormalizeData(object = pbmc, normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000)</code></pre>
<p>We are correcting for the differing read counts between cells. Example, a cell could have more total expression than another cell but both could have similar gene expression proportion. Count data is bad so we do a log transformation.</p>
<p>Seurat normalizes the gene expression for each cell by the total expression. Then multiplies this by 10,000 and then logs the result.</p>
</div>
<div id="detection-of-variable-genese-across-the-single-cells" class="section level2">
<h2><strong>Detection of variable genese across the single cells</strong></h2>
<p>We are only interested in genes that have different expression as similarly expressed genes across cells are uninteresting.</p>
<pre class="r"><code>pbmc &lt;- FindVariableGenes(object = pbmc, mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, x.high.cutoff = 3, y.cutoff = 0.5)</code></pre>
<p>One approach would have been to calculate the variance of individual genes across cells and look at the highest variances first. In our graph, “Dispersion” means variance.</p>
<p>Seurat’s approach: they make bins of mean expression and look at the variance within those bins. This is because variance can depend on the mean (e.g. high mean value may lead to high variance value).</p>
<p>y.cutoff = 0.5 refers to half a standard deviation away from the mean.</p>
<pre class="r"><code>length(x = pbmc@var.genes)</code></pre>
<p>This is the amount of variable genes that we have found.</p>
</div>
<div id="scaling-the-data-and-removing-unwanted-sources-of-variation" class="section level2">
<h2><strong>Scaling the data and removing unwanted sources of variation</strong></h2>
<p>We want to remove technical noise and also biological variation (e.g. cell cycle stage). These are “uninteresting” sources of variation.</p>
<p>The term “regress out” can be understood <a href="https://stats.stackexchange.com/questions/117840/how-to-regress-out-some-variables">here</a></p>
<pre class="r"><code>pbmc &lt;- ScaleData(object = pbmc, vars.to.regress = c(&quot;nUMI&quot;, &quot;percent.mito&quot;))</code></pre>
<p>This regression step is still a little confusing. Why are we regressing on nUMI and percent.mito?</p>
</div>
<div id="perform-linear-dimensional-reduction" class="section level2">
<h2><strong>Perform linear dimensional reduction</strong></h2>
<p>We are doing PCA here.</p>
<pre class="r"><code>pbmc &lt;- RunPCA(object = pbmc, pc.genes = pbmc@var.genes, do.print = TRUE, pcs.print = 1:5, genes.print = 5)</code></pre>
<p>Seurat can visualise PCA by printing or plotting (VizPCA, PCAPlot).</p>
<pre class="r"><code>PrintPCA(object = pbmc, pcs.print = 1:5, genes.print = 5, use.full = FALSE)
VizPCA(object = pbmc, pcs.use = 1:2)
PCAPlot(object = pbmc, dim.1 = 1, dim.2 = 2)</code></pre>
<p><del>Yeah, I think I definetly have different data from the example in the tutorial. I don’t know why though.</del></p>
<pre class="r"><code>pbmc &lt;- ProjectPCA(object = pbmc, do.print = FALSE)</code></pre>
<p>I don’t really understand what this is for.</p>
<p>It can also do PCA heatmaps.</p>
<pre class="r"><code>PCHeatmap(object = pbmc, pc.use = 1, cells.use = 500, do.balanced = TRUE, label.columns = FALSE)</code></pre>
<p>When I ran it, I got a warning saying ‘“dimTitle” is not a graphical parameter’.</p>
<pre class="r"><code>PCHeatmap(object = pbmc, pc.use = 1:12, cells.use = 500, do.balanced = TRUE, label.columns = FALSE, use.full = FALSE)</code></pre>
</div>
<div id="determine-statistically-significant-principal-components" class="section level2">
<h2><strong>Determine statistically significant principal components</strong></h2>
<pre class="r"><code>pbmc &lt;- JackStraw(object = pbmc, num.replicate = 100, display.progress = FALSE)
JackStrawPlot(object = pbmc, PCs = 1:12)</code></pre>
<pre class="r"><code>PCElbowPlot(object = pbmc)</code></pre>
</div>
<div id="cluster-the-cells" class="section level2">
<h2><strong>Cluster the cells</strong></h2>
<pre class="r"><code>pbmc &lt;- FindClusters(object = pbmc, reduction.type = &quot;pca&quot;, dims.use = 1:10, resolution = 0.6, print.output = 0, save.SNN = TRUE)

PrintFindClustersParams(object = pbmc)</code></pre>
<p>Seurat’s clustering method is as follows (I think): construct KNN (K-nearest neighbor) graph based on the euclidean distance in PCA, refine using Jaccard similarity and then Louvain algorithm (?).</p>
</div>
<div id="non-linear-dimensional-reduction-tsne" class="section level2">
<h2><strong>Non-linear dimensional reduction (tSNE)</strong></h2>
<pre class="r"><code>pbmc &lt;- RunTSNE(object = pbmc, dims.use = 1:10, do.fast = TRUE)

TSNEPlot(object = pbmc)</code></pre>
<p>Looks pretty. More about t-SNE (t-distributed stochastic neighbor embedding) <a href="https://distill.pub/2016/misread-tsne/">here</a>.</p>
<p>You can save this by using saveRDS()</p>
<pre class="r"><code>saveRDS(pbmc, file = &quot;../output/pbmc_tutorial.rds&quot;)</code></pre>
</div>
<div id="finding-differentially-expressed-genes-cluster-biomarkers" class="section level2">
<h2><strong>Finding differentially expressed genes (cluster biomarkers)</strong></h2>
<p>Seurat can find clusters using differential expression, FindAllMarkers(). min.pct is the minimum percentage the gene requires to be detected. Setting this to zero will make the function test a large number of genes that are unlikely to be differential (lot of compute time needed). max.cells.per.ident also speeds things up but I don’t quite get why.</p>
<pre class="r"><code>cluster1.markers &lt;- FindMarkers(object = pbmc, ident.1 = 1, min.pct = 0.25)
print(x = head(x = cluster1.markers, n = 5))

cluster5.markers &lt;- FindMarkers(object = pbmc, ident.1 = 5, ident.2 = c(0, 3), min.pct = 0.25)
print(x = head(x = cluster5.markers, n = 5))

pbmc.markers &lt;- FindAllMarkers(object = pbmc, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
pbmc.markers %&gt;% group_by(cluster) %&gt;% top_n(2, avg_logFC)

VlnPlot(object = pbmc, features.plot = c(&quot;MS4A1&quot;, &quot;CD79A&quot;))

# raw UMI counts included
VlnPlot(object = pbmc, features.plot = c(&quot;NKG7&quot;, &quot;PF4&quot;), use.raw = TRUE, y.log = TRUE)

FeaturePlot(object = pbmc, features.plot = c(&quot;MS4A1&quot;, &quot;GNLY&quot;, &quot;CD3E&quot;, &quot;CD14&quot;, &quot;FCER1A&quot;, &quot;FCGR3A&quot;, &quot;LYZ&quot;, &quot;PPBP&quot;, &quot;CD8A&quot;), cols.use = c(&quot;grey&quot;, &quot;blue&quot;), reduction.use = &quot;tsne&quot;)</code></pre>
<p>Ok, that took really long. The FindAllMarkers for the pbmc.markers took ~5mins.</p>
<p>I don’t understand the Vln plots.</p>
<p>The tutorial mentions alternate test methods can be used here (e.g. ROC test) and it can be changed using test.use.</p>
<p>A heatmap can also be done.</p>
<pre class="r"><code>top10 &lt;- pbmc.markers %&gt;% group_by(cluster) %&gt;% top_n(10, avg_logFC)

DoHeatmap(object = pbmc, genes.use = top10$gene, slim.col.label = TRUE, remove.key = TRUE)</code></pre>
</div>
<div id="assigning-cell-type-identity-to-clusters" class="section level2">
<h2><strong>Assigning cell type identity to clusters</strong></h2>
<p>Ok, I can see that he has given an ID to each cell type in the tutorial. But I think I have different data to him. So, I’m not sure I’m able to make anything useful here. I can still try with some of the parameters left out:</p>
<pre class="r"><code>pbmc@ident &lt;- plyr::mapvalues(x = pbmc@ident)
TSNEPlot(object = pbmc, do.label = TRUE, pt.size = 0.5)</code></pre>
<p>Yeah this comes up with an error saying the argument “from” is missing. So I definetly need to made the cluster ID number to the cluster name.</p>
<p>If I try it with the same IDs and cell types in the tutorial:</p>
<pre class="r"><code>current.cluster.ids &lt;- c(0, 1, 2, 3, 4, 5, 6, 7)
new.cluster.ids &lt;- c(&quot;CD4 T cells&quot;, &quot;CD14+ Monocytes&quot;, &quot;B cells&quot;, &quot;CD8 T cells&quot;, &quot;FCGR3A+ Monocytes&quot;, &quot;NK cells&quot;, &quot;Dendritic cells&quot;, &quot;Megakaryocytes&quot;)
pbmc@ident &lt;- plyr::mapvalues(x = pbmc@ident, from = current.cluster.ids, to = new.cluster.ids)
TSNEPlot(object = pbmc, do.label = TRUE, pt.size = 0.5)</code></pre>
<p>Some of the cell types seem to be missing. I still think it’s because I have different data from the tutorial but I may have made some small mistake along the way I’m not sure.</p>
</div>
<div id="further-subdivisions-within-cell-types" class="section level2">
<h2><strong>Further subdivisions within cell types</strong></h2>
<pre class="r"><code>pbmc &lt;- StashIdent(object = pbmc, save.name = &quot;ClusterNames_0.6&quot;)

pbmc &lt;- FindClusters(object = pbmc, reduction.type = &quot;pca&quot;, dims.use = 1:10, resolution = 0.8, print.output = FALSE)</code></pre>
<p>The output told me to set force.recalc to TRUE. But I think the tutorial ignores the warning and moves on.</p>
<pre class="r"><code>pbmc &lt;- FindClusters(object = pbmc, reduction.type = &quot;pca&quot;, dims.use = 1:10, resolution = 0.8, print.output = FALSE, force.recalc = TRUE)</code></pre>
<pre class="r"><code>plot1 &lt;- TSNEPlot(object = pbmc, do.return = TRUE, no.legend = TRUE, do.label = TRUE)
plot2 &lt;- TSNEPlot(object = pbmc, do.return = TRUE, group.by = &quot;ClusterNames_0.6&quot;, 
    no.legend = TRUE, do.label = TRUE)
plot_grid(plot1, plot2)</code></pre>
<pre class="r"><code>tcell.markers &lt;- FindMarkers(object = pbmc, ident.1 = 0, ident.2 = 1)

FeaturePlot(object = pbmc, features.plot = c(&quot;S100A4&quot;, &quot;CCR7&quot;), cols.use = c(&quot;green&quot;, &quot;blue&quot;))</code></pre>
<p>We can save our work for now by:</p>
<pre class="r"><code>pbmc &lt;- SetAllIdent(object = pbmc, id = &quot;ClusterNames_0.6&quot;)
saveRDS(pbmc, file = &quot;../output/pbmc3k_final.rds&quot;)</code></pre>
<p>I understand the saveRDS but I don’t know why we have to do the SetAllIdent</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
