<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Denny Shin" />

<meta name="date" content="2018-09-25" />

<title>zombie-lif-gene</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Learning. Thoughts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">zombie-lif-gene</h1>
<h4 class="author"><em>Denny Shin</em></h4>
<h4 class="date"><em>2018-09-25</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-09-26</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/dennyshin/Learning-Thoughts/tree/6bc7634f62bde879bdd94383ddb2c2af2e3462d7" target="_blank">6bc7634</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:   (DESKTOP-9MQETIV's conflicted copy 2018-07-26).Rhistory
    Untracked:  Learning-Thoughts.Rproj
    Untracked:  data/4k PBMCs from a Healthy Donor/
    Untracked:  data/filtered_gene_bc_matrices/
    Untracked:  docs/figure/
    Untracked:  output/pbmc3k_final.rds
    Untracked:  output/pbmc_tutorial.rds

Unstaged changes:
    Deleted:    Learning. Thoughts.Rproj
    Modified:   analysis/articles-read-2018.Rmd
    Modified:   analysis/long-read-seq-CACNA1C.Rmd
    Modified:   analysis/single-cell-RNAseq.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/dennyshin/Learning-Thoughts/blob/6bc7634f62bde879bdd94383ddb2c2af2e3462d7/analysis/zombie-lif-gene.Rmd" target="_blank">6bc7634</a>
</td>
<td style="text-align:left;">
Denny Shin
</td>
<td style="text-align:left;">
2018-09-26
</td>
<td style="text-align:left;">
added 4kpbmc and zombie lif article
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="very-short-summary" class="section level1">
<h1><strong>Very short summary</strong></h1>
<p>Large origanisms should have more risk to cancer. But evidence shows otherwise (Peto’s paradox).</p>
<p>LIF gene in elephants is upregulated in response to DNA damage which reduces the risk of cancer. The LIF6 induces apoptosis in damaged cells.</p>
</div>
<div id="introduction" class="section level1">
<h1><strong>Introduction</strong></h1>
<p>Larger body and longer life span should equal higher risk of cancer. This is consistent within species but not across different species (Peto’s paradox).</p>
<p>oncogene = gene that has potential to cause cancer</p>
<p>Interleukin-6 class cytokine leukemia inhibitory factor (LIF) can be a tumor suppressor or and oncogene depending on context.</p>
<p>LIF6, which is similar to LIF-T, induces apoptosis when overexpresse.</p>
</div>
<div id="results" class="section level1">
<h1><strong>Results</strong></h1>
<div id="repeated-segmental-duplications-increased-lif-copy-number-in-paenungulates" class="section level4">
<h4><strong>Repeated SEgmental Duplications Increased LIF Copy NUmber in Paenungulates</strong></h4>
<p>Paenungulata (Manatee, rock hyrax, african elephant) have hight copy number variation than the other species studied.</p>
<ul>
<li>Why is copy number variation of the LIF gene significant here?</li>
</ul>
<blockquote>
<p>None of the duplicate LIF genes includes the 50-UTR, coding exon 1, or a paired low complexity (CGAG)n/CT-rich repeat common to the canonical LIF genes in elephant, hyrax, manatee, tenrec, and armadillo</p>
</blockquote>
<p>Figure 2 indicates that:</p>
<blockquote>
<p>the additional LIF genes result from repeated rounds of segmental duplication</p>
</blockquote>
<ul>
<li>How to look at Figure 2?</li>
</ul>
</div>
<div id="duplicate-lif-genes-are-structurally-similar-to-the-lif-t" class="section level4">
<h4><strong>Duplicate LIF genes are structurally similar to the LIF-T</strong></h4>
<p>The duplicates of the LIF gene keep the pro-apoptotic feature of LIF-T. This feature is from the leucine and/or isoleucine repeat in the polypeptide sequence.</p>
</div>
<div id="elephant-lif6-is-upregulated-by-tp53-in-response-to-dna-damage" class="section level4">
<h4><strong>Elephant LIF6 is upregulated by TP53 in Response to DNA damage</strong></h4>
<p>The main thing that is interesting is whether the expansion of LIF genes plays a role in the enhance cancer resistance. If it is, then one or more LIF genes should be transcribed.</p>
<p>IN the asian elephant dataset, the LIF6 transcript expression was extremely low (this is actually expected? because it is a pro-apoptotic gene).</p>
<p>The investigation then turns to TP53, which is known regulate LIF in response to DNA damage. Elephant LIF6 was found to have TP53 binding sites.</p>
<p>They then treated african elephant cells with a DNA-damaging agent (doxurubicin = DOX or MDM2).</p>
<p>DOX = common chemotherapy medication</p>
<p>MDM2 = mediated TP53 (TP53 = tumor suppressor 53)</p>
<p>LIF1 gene did not respond to either treatment. LIF6 genes increased in expression by ~8 and ~16.</p>
<p>To test the TP53 binding site upstream of elephant LIF6: the region was cloned onto a luciferase (related to bioluminesence) vector. Since, it is unrelated, the presence of the binding site should have no effect and the evidence shows as such.</p>
<p>I think the next part is saying that they took rabbit TP53 and tested whether it binded with elephant TP53 binding sites. With DOX, MDM2 treatment, there was an increase in binding.</p>
<p>siRNA = small interfering RNA which are double stranded RNA that preven mRNA translation</p>
<p>siRNA significantly reduced TP53 transcript levels. Knockdown of TP53 did not inhibit the induction of apoptosis. Suggests that LIF6 can induce apoptosis in the absence of DNA damage similar to LIF-T.</p>
<blockquote>
<p>Thus, we conclude that elephant LIF6 is transcriptionally upregulated by TP53 in response to DNA damage and may have pro-apoptotic functions.</p>
</blockquote>
<p>How? It is not very clear to me.</p>
</div>
<div id="elephant-lif6-contributed-to-the-augmented-dna-damage-response-in-elephants" class="section level4">
<h4><strong>Elephant LIF6 contributed to the augmented DNA-damage response in elephants</strong></h4>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
