---
title: "long-read-seq-CACNA1C"
author: "Denny Shin"
date: "2018-08-08"
output: workflowr::wflow_html
---

## **Motivation**

We focus on the impact of RNA splicing on schizophrenia, more specifically, the CACNA1C gene. Why schizophrenia? Well, because schizophrenia is the disorder which most genetic loci have been identified to date (Feb 2018). Short-read RNAseq misses out on alot of the gene and splice isoforms. Also, because of read alignment it is hard to distinguish between full-length isoforms. Our motivation is to use long-red RNAseq to improve RNA splice isoform discovery.

## **CACNA1C gene**

CACNA1C encodes a voltage-gated calcium channel (VGCC) and is a leading candidate gene forpsychiatric disorders. We currently (Feb 2018) observe ~50 exons and 40 isoforms (from splicing and transcriptional mechanisms). Alternative forms of the gene shows functionally different VGCCs.

## **Method**

Oxford nanopore will be used to provide the long-read sequences while bearing in mind that these reads have higher than normal error rates. To compensate, RT-PCR is combined with nanopore technolofy. 

There is a lot of terminology that I don't understand but I think I get the general idea of what they're doing. After RNA extraction (which I have no clue about) they do RT-PCR followed by nanopore sequencing. They map the reads and annotate transcripts. From this they find the expression levels. They also find new exons and exon junctions (the joining place of two exons on the mature mRNA). Lastly, they mention the effect of amplification of full-length genes but I don't really follow. Note: RIN = RNA integrity number.

## **Results**

#### **Long-range seq reveal new things**

Specifically, gene isoforms and exons. The new exons were validated using PCR and Sanger sequencing. Results also show high expression of a previously known transcript as expected. It validates the new exons and isoforms and also suggests that current annotations are missing many or represent repeats/errors. 

#### **Impact on protein sequence**

Recall: CACNA1C encodes a VGCC. VGCC contains a pore, which are made of transmembrane repeats. It is known that disruptions to these membrane lead to a non-functioning protein. Thus, they looked for isoforms where the membrane coding regions are impacted. Long read sequencing has proved useful once again by being able to reliably identify microindels (given we have enough reads). The article also talks about the impact on the differen domains and channel properties of the protein.

#### **Expression across different brain regions (tissues)**

They found that individuals had no impact on gene expression but brain tissue did. Different tissues also showed different dominance of gene isoforms i.e. different tissues had different proportions of gene isoforms. 

#### **Problems with long-red sequencing**

High quality of RNA is needed for RT-PCR and long-read sequences. Post-mortem samples are often degraded. Article suggest a minimum RINe (?) value of 6 or 7 depending on intended use. 

## **My thoughts**

So, always take my own thoughts with a full bag of salt because I know much less than any of these people. It is nice that we identified more transcripts but how do we compare the quality of these findings versus other methods? The article suggests very boldly that many other isoforms are repeats and errors but I am not really convinced. There is good reason to think this but I think that further testing and improvement of long-read sequences should be done before we reach this conclusion. Regardless, it is still exciting to think that long-reads are not far away from being used reliably for a variety of studies.
